File: lib/src/main/java/com/sd/lib/cache/handler/DoubleHandler.java
package com.sd.lib.cache.handler;

import com.sd.lib.cache.CacheInfo;

/**
 * Double处理类
 */
public class DoubleHandler extends BaseCacheHandler<Double> {
    public DoubleHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(Double value) {
        return value.toString().getBytes();
    }

    @Override
    protected Double byteToValue(byte[] bytes, Class<?> clazz) {
        return Double.valueOf(new String(bytes));
    }

    @Override
    protected String getKeyPrefix() {
        return "double_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/DoubleHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * Double处理类
 */
internal class DoubleHandler(info: CacheInfo) : BaseCacheHandler<Double>(info, "double") {
    override fun encodeToByteImpl(value: Double): ByteArray {
        return value.toString().toByteArray()
    }

    override fun decodeFromByteImpl(bytes: ByteArray, clazz: Class<*>?): Double {
        return String(bytes).toDouble()
    }
}